crates/zk-por-core/src/circuit_registry/registry.rs

```diff
@@ -0,0 +1,167 @@
+use plonky2::plonk::{
+    circuit_data::{CircuitConfig, CircuitData},
+    proof::ProofWithPublicInputs,
+};
+
+use crate::{
+    account::Account,
+    // circuit_config::{BATCH_SIZE, RECURSIVE_FACTOR, ASSET_NUM},
+    merkle_sum_prover::{
+        circuits::{
+            account_circuit::AccountTargets, merkle_sum_circuit::build_merkle_sum_tree_circuit,
+        },
+        prover::MerkleSumTreeProver,
+    },
+    recursive_prover::prover::RecursiveProver,
+    recursive_prover::recursive_circuit::{build_recursive_n_circuit, RecursiveTargets},
+};
+
+use plonky2_field::types::Field;
+use rand::Rng;
+
+use crate::types::{C, D, F};
+// use once_cell::sync::Lazy;
+
+pub struct CircuitRegistry<const RECURSIVE_FACTOR: usize> {
+    batch_circuit: (CircuitData<F, C, D>, Vec<AccountTargets>),
+    empty_batch_proof: ProofWithPublicInputs<F, C, D>,
+    recursive_circuits_and_empty_proofs: Vec<(
+        (CircuitData<F, C, D>, RecursiveTargets<RECURSIVE_FACTOR>),
+        ProofWithPublicInputs<F, C, D>,
+    )>,
+}
+
+fn gen_empty_accounts(num_accounts: usize, num_assets: usize) -> (Vec<Account>, u32, u32) {
+    let mut accounts: Vec<Account> = Vec::new();
+    let mut rng = rand::thread_rng(); // Create a random number generator
+    let mut equity_sum = 0;
+    let mut debt_sum = 0;
+    for _ in 0..num_accounts {
+        let mut equities = Vec::new();
+        let mut debts = Vec::new();
+        for _ in 0..num_assets {
+            let equity = 0;
+            let debt = 0;
+            equity_sum += equity;
+            debt_sum += debt;
+            equities.push(F::from_canonical_u32(equity));
+            debts.push(F::from_canonical_u32(debt));
+        }
+        let mut bytes = [0u8; 32]; // 32 bytes * 2 hex chars per byte = 64 hex chars
+        rng.fill(&mut bytes);
+        let account_id = bytes.iter().map(|byte| format!("{:02x}", byte)).collect::<String>();
+        accounts.push(Account { id: account_id, equity: equities, debt: debts });
+    }
+    (accounts, equity_sum, debt_sum)
+}
+
+impl<const RECURSIVE_FACTOR: usize> CircuitRegistry<RECURSIVE_FACTOR> {
+    pub fn init(
+        batch_size: usize,
+        asset_num: usize,
+        batch_circuit_config: CircuitConfig,
+        recursive_level_configs: Vec<CircuitConfig>,
+    ) -> Self {
+        let init_start = std::time::Instant::now();
+
+        let start = std::time::Instant::now();
```

Comment:
can have a macro like this
macro_rules! time_it {
    ($name:expr, $block:block) => {{
        use std::time::Instant;

        let start = Instant::now();
        let result = $block; // Execute the code block
        let duration = start.elapsed();

        println!("Time elapsed for {}: {:?}", $name, duration);

        result // Return the result of the block
    }};
}

Reply:
try to use this macro. Got some unexpected bug.